common/src/lib.rs

```diff
@@ -191,6 +195,12 @@ pub trait ValueTree {
     /// the `key` belongs to a subtree originating at such "orphan" node.
     fn get_origin(key: Self::Key) -> Result<Option<Self::ExternalOrigin>, Self::Error>;
 
+    /// The id of external node for a key, if the latter exists, `None` otherwise.
+    ///
+    /// Error occurs if the tree is invalidated (has "orphan" nodes), and the node identified by
+    /// the `key` belongs to a subtree originating at such "orphan" node.
+    fn get_origin_key(key: Self::Key) -> Result<Option<Self::ExternalOrigin>, Self::Error>;
```

Comment:
The signature of this function is identical to the one above - it's natural to think they are doing the same thing.
If this function returns the key of a value tree root node then it should return `Self::Key`, should it not?
The `Self::ExternalOrigin` type is supposed to identify the external owner of the whole tree - that is some actor in actor space. As opposed to the `Self::Key` type which defines the key of a node (including the root node).
The fact they are both represented in the `pallet-gas` as `H256` is a bit confusing though.

Reply:
Yeah, that's true, refactored